doc/specs/#1595 - Suggestions UI/Suggestions-UI.md

```diff
@@ -0,0 +1,734 @@
+* If the user hasn't enabled shell completion, we could add text to the
+  `commandHistory` or `directoryHistory` menus to inform the user how they could
+  go enable shell integration. We already have a docs page dedicated to this, so
+  we could start by linking to that page. More notes on this in [Automatic shell
+  integration](#Automatic-shell-integration).
+* Maybe there could be a per-profile setting for automatic suggestions after
+  some timeout. Like, as you type, a menu version of the Suggestions UI appears.
+  So you could just start typing `git c`, and it would automatically give you a
+  menu with suggestions, implicitly using the typed command as the "filter".
+  * Maybe we could do this as an `implicit` property on the `suggestions` action
+
+
+#### Description Tooltips
+
+> **Note**: _This is left as a future consideration for the initial draft of
+> this spec. I'd like to flesh out [shell-driven autocompletion] more before
+> committing any plans here._
+
+It would be beneficial for the Suggestions UI to display additional context to
+the user. Consider a extension that provides some commands for the user, like a
+hypothetical "Docker" extension. The extension author might be able to give the
+commands simplified names, but also want to expose a more detailed description
+of the commands to the user.
+
+Or consider the Suggestions UI when invoked by [shell-driven autocompletion].
+The shell might want to provide help text to the user with each of the
+suggestions. This would allow a user to browse through the suggestions that they
+might not know about, and learn how they work before committing to one.
+
+Only the help text for the currently hovered command should be presented to the
+user. To support this kind of UX, we'll add an optional flyout of some sort to
+display with the Suggestions UI. This flyout will only appear if there's more
+information provided to the Terminal.
+
+This might be in the form of a `TeachingTip`, as in this example:
+
+![TeachingTip with description](https://user-images.githubusercontent.com/18356694/222244568-243a6482-92d9-4c3c-bffc-54ad97f01f69.gif)
+
+Actions in the settings could also accept an optional `description` property, to
+specify the string that would be presented in that flyout.
+
+#### Automatic shell integration
+
+A large portion of these features all rely on shell integration being enabled by
+the user. However, this is not a trivial thing for the Terminal to do on behalf
+of the user. Shell integration relies on changes to the user's shell config. If
+the Terminal were to try and configure those itself, we may accidentally destroy
+configuration that the user has already set up. Hence why the Terminal can't
+just have a "Light up all the bells and whistles" toggle in the Settings UI.
+
+This is a non-trivial problem to solve, so it is being left as a future
+consideration. It deserves its own spec to sort out how we should expose this to
+users and safely implement it.

```

Comment:
This section may belong in the broader Shell Integration spec rather than in here.